### PR TITLE
Renamed files according to classnames, added extension method for eas…

### PR DESCRIPTION
**Pull request by fredericbirke**
- Renamed files according to classnames
 - added extension method for easier applying of ExternalDirectiveType in CodeFirst
 - added two more testcases for ExternalDirective to ensure corrent application of it in schema first and CodeFirst

### CODE REVIEW

File: src/HotChocolate/ApolloFederation/test/ApolloFederation.Tests/Directives/ExternalDirectiveTests.cs [hunk 1]
```diff
@@ -29,6 +31,57 @@ public void AddExternalDirective_EnsureAvailableInSchema()
...
+                            .Type<StringType>()
+                            .External()
+                    ));
+
+                    t.RegisterDirective<ExternalDirectiveType>();
+
+                    t.Use(next => context => default(ValueTask));
+                });
+
+            ObjectType query = schema.GetType<ObjectType>("Query");
```
Comment (michaelstaib) [inline src/HotChocolate/ApolloFederation/test/ApolloFederation.Tests/Directives/ExternalDirectiveTests.cs:57]: where is the assert part of this test?

File: src/HotChocolate/ApolloFederation/test/ApolloFederation.Tests/Directives/ExternalDirectiveTests.cs [hunk 2]
```diff
@@ -29,6 +31,57 @@ public void AddExternalDirective_EnsureAvailableInSchema()
...
+                    interface IQuery {
+                        field(a: Int): String
+                            @external
+                    }
+                    ")
+                .AddDirectiveType<ExternalDirectiveType>()
+                .Use(next => context => default(ValueTask))
+                .Create();
+
+            InterfaceType queryInterface = schema.GetType<InterfaceType>("IQuery");
```
Comment (michaelstaib) [inline src/HotChocolate/ApolloFederation/test/ApolloFederation.Tests/Directives/ExternalDirectiveTests.cs:83]: where is the assert part of this test?

File: src/HotChocolate/ApolloFederation/test/ApolloFederation.Tests/Directives/ExternalDirectiveTests.cs (hunk 1 above)
Comment (michaelstaib) [reply src/HotChocolate/ApolloFederation/test/ApolloFederation.Tests/Directives/ExternalDirectiveTests.cs:57]: The test should always have proper asserts ... Assert.Collection for instance

Comment (michaelstaib) [reply src/HotChocolate/ApolloFederation/test/ApolloFederation.Tests/Directives/ExternalDirectiveTests.cs:57]: As to why, a single exception is not explicit enough and might be more difficult to read in a test result file.

Comment (fredericbirke) [reply src/HotChocolate/ApolloFederation/test/ApolloFederation.Tests/Directives/ExternalDirectiveTests.cs:57]: Replaced the collectin.single with Assert.Collection

File: src/HotChocolate/ApolloFederation/test/ApolloFederation.Tests/Directives/ExternalDirectiveTests.cs (hunk 2 above)
Comment (fredericbirke) [reply src/HotChocolate/ApolloFederation/test/ApolloFederation.Tests/Directives/ExternalDirectiveTests.cs:86]: Replaced the collectin.single with Assert.Collection

**Review by michaelstaib — APPROVED** [review]

**Pull request merged**